src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/FileStreamReadLightUp.cs

```diff
@@ -156,7 +156,7 @@ IntPtr overlapped
             );
 
             // older Windows systems:
-            [DllImport(@"kernel32.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
+            [DllImport(@"api-ms-win-core-file-l1-1-0.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
```

Comment:
I can tell you one reason.   A client wanted to use bluetooth as a
parameter boundary to turn off networking for surface tablets.
In UWP you can detect bluetooth but not turn off networking. but in regular
windows on tablet win8/10 you can not detect bluetooth.(seriously try it)
 so a file somewhere would be a sentinel to turn on off networking.

Sincerely,
J. John Mele
631 615 7213

linkedin profile: http://www.linkedin.com/in/joemele
Youseful Software
www.youseful.com

On Wed, Apr 6, 2016 at 1:15 PM, Nick Guerrera <notifications@github.com>
wrote:

> In
> src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/FileStreamReadLightUp.cs
> <https://github.com/dotnet/corefx/pull/7532#discussion_r58744853>:
>
> > @@ -156,7 +156,7 @@ IntPtr overlapped
> >              );
> >
> >              // older Windows systems:
> > -            [DllImport(@"kernel32.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
> > +            [DllImport(@"api-ms-win-core-file-l1-1-0.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
>
> FWIW, at least one user was using it in a windows 8 store app (no idea
> why), but a bug was reported via nugget feedback and subsequently fixed. I
> recall that ReadFile via kernel32.dll was actually allowed on win8 when I
> checked with @ericstj <https://github.com/ericstj>. Did that change with
> UWP?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/corefx/pull/7532/files/df35c1146cde41cbb605a344fbf085fa8eba5d61#r58744853>
>